base/base.py

```diff
@@ -17,8 +21,51 @@ def read_app_data(self, file_name, file_path=None):
         	Returns:
         	    Parsed data in dictionary
         '''
-        if td_dir is None:
-            td_dir = self.test["case_td_dir"]
+        if file_name is None or class_name is None:
+            print('Cannot get the test data file name') 
+            raise Exception
+        else:
+            stack = inspect.stack()[1]
+            if dir_name is None:
+                dir_name = os.path.basename(stack[1])
+            if fn_name is None:
+                fn_name = stack[3]          
+            file = os.getcwd() + '/applications/sample/testdata' + '/' + dir_name + '/' + class_name +'/' + fn_name + '/' + file_name
+
+            #Check if test data file exists at the last level , if it doesnt move back to 
+            # eg. File name : toothless/fb-apps/applications/sample/testdata/test_class.py/TestClass/test_three/info.json 
+            print('File : {} '.format(file))          
+            if os.path.isfile(file):
+                print('Found file '.format(file))
+            else:
+                file = ""
+                file = os.getcwd() + '/applications/sample/testdata' + '/' + dir_name + '/' + class_name +'/' + file_name
+                print('File : {} '.format(file)) 
+                if os.path.isfile(file):
+                    print('Found file '.format(file))
+                else:
+                    file = ""
+                    file = os.getcwd() + '/applications/sample/testdata' + '/' + dir_name + '/' + file_name
+                    print('File : {} '.format(file)) 
+                    if os.path.isfile(file):
```

Comment:
Instead of checking these with nested if, better way would be to try opening the file with each value. If the file does not exists, we ll get an IOException. Catch the IOException and try the next line, and so on.
Finally, if all three tries threw exception, then raise and assert